### PR TITLE
feat: institution selector

### PR DESCRIPTION
**Pull request by 01001110J**
# Description

In this PR is added the drop down selector for multiple institutions

This PR resolves [PADV-1505](https://agile-jira.pearson.com/browse/PADV-1505)

## Change log
- Added new drop down.
- The institution id is preserved on the url.
- Updated unit test.

### How to test

Clone the Institution Portal MFE
```Bash
     git clone <url>
```
Install the packages 📦.
```Bash
     npm i
```
Start project.
```Bash
     npm start
```

Finally, go to http://localhost:1980/ and use the different current links to navigate around, you should be able to have the institution Id in the ulr in case to have multiple institutions linked to your account..

### CODE REVIEW

File: src/hooks/index.jsx
Comment (AuraAlba) [inline src/hooks/index.jsx:1]: This hook could be moved to the library

File: src/helpers/index.jsx [hunk 1]
```diff
@@ -27,3 +29,38 @@ export const emailValidationMessages = (emailList) => {
...
+/**
+ * Transforms an array of options to the format required for select components.
+ *
+ * @param {Array} options - The array of options to be formatted.
+ * @param {string} options[].name - The name of the option.
+ * @param {number|string} options[].id - The ID of the option.
+ *
+ * @returns {Array|null} The formatted options array or null if there's an error.
+ */
+export const formatSelectOptions = (options) => {
```
Comment (AuraAlba) [inline src/helpers/index.jsx:42]: Move this to the helpers in library

File: src/features/Main/InstitutionSelector/index.jsx
Comment (AuraAlba) [inline src/features/Main/InstitutionSelector/index.jsx:1]: In filters, students and classes, the select options is not loaded correctly when you change the institutionId, also when you apply filters is not send the institution Id

**Review by AuraAlba — CHANGES_REQUESTED** [review]

File: src/features/Main/InstitutionSelector/index.jsx
Comment (01001110J) [reply src/features/Main/InstitutionSelector/index.jsx:1]: > In filters, students and classes, the select options is not loaded correctly when you change the institutionId

Could you please tell me how to reproduce the issue? I been tried on postman and everything looks good, but maybe I'm testing wrong.

File: src/hooks/index.jsx
Comment (01001110J) [reply src/hooks/index.jsx:1]: Not sure about it, it involve to install redux  in the lib, which is not bad but, keep the same lib in 3 different repos, idk it does not convince to me, also is a pretty simple hook and the way in how we access to the state is different in both repositories.

File: src/features/Main/InstitutionSelector/index.jsx
Comment (AuraAlba) [reply src/features/Main/InstitutionSelector/index.jsx:1]: When you are in students filters you see the courses options, but when you change the institution the options are the same and should be updated, same for classes filters. Let me know if you can't see it

Comment (01001110J) [reply src/features/Main/InstitutionSelector/index.jsx:1]: Fixed

Comment (AuraAlba) [reply src/features/Main/InstitutionSelector/index.jsx:1]: In filters I lost the institution id query when I click reset

Comment (01001110J) [reply src/features/Main/InstitutionSelector/index.jsx:1]: Fixed @AuraAlba

**Review by AuraAlba — APPROVED** [review]

**Review by anfbermudezme — APPROVED** [review]

**Pull request merged**